### ISSUE TITLE
Syntactically invalid result on Esqueleto code

### ISSUE THREAD

**Issue by MaxGabriel** [lspitzner/brittany#101]
Running `brittany /Users/maximiliantagher/Documents/Mercury/hs/mercury-web-backend/src/Handler/Home.hs` gives me: "ERROR: brittany pretty printer returned syntactically invalid result.", even though the code compiles fine.

Source file:

```haskell
module Handler.Home where

import Import
import qualified Database.Esqueleto      as E
import           Database.Esqueleto      ((^.))
import Data.String.Conversions

getHomeR :: Handler Value
getHomeR = do
    result <- runDB $
        E.select $
        E.from $ \(u `E.InnerJoin` ou `E.InnerJoin` o) -> do
        E.on (ou ^. OrganizationUserOrganizationId E.==. o ^. OrganizationId)
        E.on (u ^. UserId E.==. ou ^. OrganizationUserUserId)
        return (u, ou, o)
    putStrLn $ cs $ show result
    returnJson [1 :: Integer]

```

If you remove the Esqueleto code (the `result <- ...` block of code), brittany runs successfully.

Version:

```
maximiliantagher@Maximilians-MacBook-Pro ~> brittany --version
brittany version 0.9.0.0
Copyright (C) 2016-2017 Lennart Spitzner
There is NO WARRANTY, to the extent permitted by law.
```

OS: macOS High Sierra 10.13.2
GHC: Should be 8.2.2 since brittany was installed with LTS 10.

Installation method: `stack install brittany --resolver lts-10.0`

I also tried with my older version of brittany, which was 0.8.something, and compiled with GHC 8.0.x.

Brittany configuration:

```yaml
conf_debug:
  dconf_roundtrip_exactprint_only: false
  dconf_dump_bridoc_simpl_par: false
  dconf_dump_ast_unknown: false
  dconf_dump_bridoc_simpl_floating: false
  dconf_dump_config: false
  dconf_dump_bridoc_raw: false
  dconf_dump_bridoc_final: false
  dconf_dump_bridoc_simpl_alt: false
  dconf_dump_bridoc_simpl_indent: false
  dconf_dump_annotations: false
  dconf_dump_bridoc_simpl_columns: false
  dconf_dump_ast_full: false
conf_forward:
  options_ghc:
  - -XLambdaCase
  - -XMultiWayIf
  - -XGADTs
  - -XPatternGuards
  - -XViewPatterns
  - -XRecursiveDo
  - -XTupleSections
  - -XExplicitForAll
  - -XImplicitParams
  - -XQuasiQuotes
  - -XTemplateHaskell
  - -XBangPatterns
  - -XFlexibleContexts
  - -XOverloadedStrings
  - -XRecordWildCards
  - -XMultiParamTypeClasses
  - -XTypeFamilies
  - -XFlexibleInstances
  - -XGeneralizedNewtypeDeriving
conf_errorHandling:
  econf_ExactPrintFallback: ExactPrintFallbackModeInline
  econf_Werror: false
  econf_omit_output_valid_check: false
  econf_produceOutputOnErrors: false
conf_preprocessor:
  ppconf_CPPMode: CPPModeAbort
  ppconf_hackAroundIncludes: false
conf_version: 1
conf_layout:
  lconfig_altChooser:
    tag: AltChooserBoundedSearch
    contents: 3
  lconfig_importColumn: 60
  lconfig_alignmentLimit: 30
  lconfig_indentListSpecial: true
  lconfig_indentAmount: 2
  lconfig_alignmentBreakOnMultiline: true
  lconfig_cols: 160
  lconfig_indentPolicy: IndentPolicyFree
  lconfig_indentWhereSpecial: true
  lconfig_columnAlignMode:
    tag: ColumnAlignModeMajority
    contents: 0.7
```

If you'd like, I can try to reproduce in a standalone file that you could compile yourself (this is just a random file in a web server I'm working on).

**Comment by eborden** [comment]
As far as I can tell this is an issue with infix type constructors.

This explodes:
```
getHomeR = E.from $ \(u `E.InnerJoin` ou) -> return (u, ou)
```

This does not:
```
getHomeR = E.from $ \(E.InnerJoin u ou) -> return (u, ou)
```

**Comment by eborden** [comment]
produced with `--omit-output-check`

```
getHomeR = \(uE.InnerJoinou) -> return (u, ou, o)

getHomeR = \(E.InnerJoin u ou) -> return (u, ou, o)
```

**Comment by eborden** [comment]
Should have a fix for this soon.

**Comment by eborden** [comment]
With the linked fix:

```
module Handler.Home where

import Import
import qualified Database.Esqueleto      as E
import           Database.Esqueleto      ((^.))
import Data.String.Conversions

getHomeR :: Handler Value
getHomeR = do
  result <-
    runDB $ E.select $ E.from $ \(u `E.InnerJoin` ou `E.InnerJoin` o) -> do
      E.on (ou ^. OrganizationUserOrganizationId E.==. o ^. OrganizationId)
      E.on (u ^. UserId E.==. ou ^. OrganizationUserUserId)
      return (u, ou, o)
  putStrLn $ cs $ show result
  returnJson [1 :: Integer]
```

**Comment by MaxGabriel** [comment]
Thank you! Do you need me to test this or anything?

**Comment by lspitzner** [comment]
The fix by eborden is published as part of 0.9.0.1.
